x-pack/plugins/timelines/public/components/t_grid/toolbar/fields_browser/index.tsx

```diff
@@ -43,6 +43,8 @@ export const StatefulFieldsBrowserComponent: React.FC<FieldBrowserProps> = ({
 
   /** all field names shown in the field browser must contain this string (when specified) */
   const [filterInput, setFilterInput] = useState('');
+
+  const [appliedFilterInput, setAppliedFilterInput] = useState('');
```

Comment:
You missed calling `setAppliedFilterInput('');` inside the _onHide_ function. This causes a bug that makes cypress tests fail, the reason is that when the fieldBrowser has an input in the search, and closes/opens the fieldBrowser, `searchInput` value is reset to '' but the `appliedFilterInput` stays defined, cypress then types the same input as before very fast and the `appliedFilterInput` does not change, so the update does not happen.
Just reseting the `appliedFilterInput` in the onHide function solves everything.